agent/taskresource/volume/dockervolume.go

```diff
@@ -53,15 +71,17 @@ type DockerVolumeConfig struct {
 
 // NewVolumeResource returns a docker volume wrapper object
 func NewVolumeResource(name string,
+	dockerVolumeName string,
```

Comment:
Are these guaranteed to be unique? Does that matter to us?

Reply:
Docker volume name for replacement "empty volume" are guaranteed to be unique (because we create it to force the scope to the running task). The Name for shared volume isn't, because it's used to refer to existing volume.